### DISCUSSION TITLE
Using Startup folder for automatic startup of application

### DISCUSSION THREAD

**Discussion by sonnichjensen** [wixtoolset/issues#6911] [Questions] [answered]
I am making an installer, which should automatically start an application when Windows starts.
This can be done by adding it to the StartUp folder in Windows ( folder in C:\ProgramData\Microsoft\Windows\Start Menu\Programs\StartUp if not under personalised )

All this compiles, and renaming the "StartUp" will cause it not to compile.
However when I run it, it does not create the autorun that I want.

How can I achieve this?

There is 
My code:

In Wix I can add those folders for shortcuts.
```
      <Directory Id="DesktopFolder" SourceName="Desktop" />
      <Directory Id="StartUpFolder" SourceName="StartUp" />
```
and the shortcut itself
```
      <Shortcut
        Id="StartUpShortcut"
        Directory="StartUpFolder"
        Name="$(var.ShortProductName)"
        WorkingDirectory="ProgramFilesFolder_blahblah"
        Target="[ProgramFilesFolder_blahblah]$(var.project.TargetFileName) /autorun"
        Advertise="no"
        Icon="myIcon.exe"
        IconIndex ="0"/>
```

**Comment by wyrdfish** [comment, marked as answer]
I would use the run registry key for this https://docs.microsoft.com/en-us/windows/win32/setupapi/run-and-runonce-registry-keys

But its not clear from your code how you are specifying the folder for the shortcut. 
Run the msi with logging to see where it is placing the shortcut.

**Reply by sonnichjensen** [reply]
Thanks
It seems to be the best option.
Now I found, that some policies on Win10+ will no allow this at all. But the registry will work
